packages/wallet/ui/src/bridge.jsx

```diff
@@ -0,0 +1,144 @@
+// @ts-check
+/* eslint-disable import/no-extraneous-dependencies */
+import '@endo/eventual-send/shim';
+import './lockdown.js';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+Error.stackTraceLimit = Infinity;
+
+/** ISSUE: where are these defined? */
+const BridgeProtocol = /** @type {const} */ ({
+  loaded: 'walletBridgeLoaded',
+  opened: 'walletBridgeOpened',
+});
+
+/**
+ * Install a dApp "connection" where messages posted from
+ * the dApp are forwarded to localStorage and vice versa.
+ *
+ * @param {{
+ *   addEventListener: typeof window.addEventListener,
+ *   parentPost: typeof window.postMessage,
+ *   t0: number,
+ * }} io
+ */
+const installDappConnection = ({ addEventListener, parentPost, t0 }) => {
+  /** @type { string } */
+  let origin;
+
+  /** @type {(key: string, value: string) => void} */
+  let setItem;
+  /** @type {string[]} */
+  const buffer = [];
+
+  console.debug('bridge: installDappConnection');
+
+  parentPost({ type: BridgeProtocol.loaded }, '*');
+  parentPost({ type: BridgeProtocol.opened }, '*');
+
+  let outIx = 0; // TODO: persist index in storage
+  /** @param {Record<string, any>} payload */
+
+  const { stringify, parse } = JSON;
+
+  addEventListener('message', ev => {
+    // console.debug('bridge: handling message:', ev.data);
+    if (
+      !ev.data ||
+      typeof ev.data.type !== 'string' ||
+      !ev.data.type.startsWith('CTP_')
```

Comment:
```suggestion
      !(ev.data?.type?.startsWith('CTP_')
```
If `ev.data.type` is defined and not a string, that's an error to fail on.

Reply:
Done